analytical_engine/apps/simple_path/all_simple_paths_context.h

```diff
@@ -0,0 +1,198 @@
+limitations under the License.
+Author: Ma JingYuan<nn9902@qq.com>
+*/
+
+#ifndef ANALYTICAL_ENGINE_APPS_SIMPLE_PATH_ALL_SIMPLE_PATHS_CONTEXT_H_
+#define ANALYTICAL_ENGINE_APPS_SIMPLE_PATH_ALL_SIMPLE_PATHS_CONTEXT_H_
+
+#include <sys/time.h>
+#include <time.h>
+#include <unistd.h>
+#include <limits>
+#include <queue>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "apps/simple_path/utils.h"
+#include "folly/dynamic.h"
+#include "folly/json.h"
+#include "grape/grape.h"
+
+#include "core/context/tensor_context.h"
+
+namespace gs {
+
+template <typename FRAG_T>
+class AllSimplePathsContext
+    : public TensorContext<FRAG_T, typename FRAG_T::oid_t> {
+ public:
+  using oid_t = typename FRAG_T::oid_t;
+  using vid_t = typename FRAG_T::vid_t;
+  using vertex_t = typename FRAG_T::vertex_t;
+
+  explicit AllSimplePathsContext(const FRAG_T& fragment)
+      : TensorContext<FRAG_T, typename FRAG_T::oid_t>(fragment) {}
+
+  void Init(grape::DefaultMessageManager& messages, oid_t source_id,
+            const std::string& targets_json,
+            int cutoff = std::numeric_limits<int>::max()) {
+    auto& frag = this->fragment();
+    this->source_id = source_id;
+    this->id_mask = frag.id_mask();
+    this->fid_offset = frag.fid_offset();
+
+    if (cutoff == std::numeric_limits<int>::max()) {
+      this->cutoff = frag.GetTotalVerticesNum() - 1;
+    } else {
+      this->cutoff = cutoff;
+    }
+
+    // init targets.
+    vid_t v;
+    std::vector<oid_t> oid_array;
+    folly::dynamic nodes_array = folly::parseJson(targets_json);
+    convert_to_oid_array(nodes_array, oid_array);
+    for (const auto& val : oid_array) {
+      if (!frag.Oid2Gid(val, v)) {
+        LOG(ERROR) << "Graph not contain vertex " << val << std::endl;
+        nodes_not_found = true;
+        break;
+      }
+      if (!targets.count(v)) {
+        targets.insert(v);
+      }
+    }
+
+    if (!frag.Oid2Gid(source_id, v)) {
+      return;
+    }
+    if (targets.count(v) || this->cutoff < 1) {
+      return;
+    }
+
+    vertex_t source;
+    bool native_source = frag.GetInnerVertex(source_id, source);
+    if (native_source) {
+      int v_size = frag.GetTotalVerticesNum();
+      VLOG(0) << "vsize: " << v_size << std::endl;
+      frag_vertex_num.resize(frag.fnum());
+      frag_vertex_num[frag.fid()] = frag.GetInnerVerticesNum();
+    }
+  }
+
+  void Output(std::ostream& os) override {
+    auto& frag = this->fragment();
+    vertex_t source;
+    bool native_source = frag.GetInnerVertex(source_id, source);
+    std::vector<vid_t> q;
+    if (native_source) {
+      std::set<vid_t> qvisit;
+      vid_t source_gid = frag.Vertex2Gid(source);
+      qvisit.insert(source_gid);
+      q.push_back(source_gid);
+      int index = find_edge_map_index(source_gid);
+      Pint_Result(index, 0, q, qvisit, os);
+    }
+  }
+
+  void Pint_Result(int from, int depth, std::vector<vid_t>& q,
```

Comment:
We DO NOT use `_` type method naming.  if public method, use upper camel case style， if private use `lower camel case`